go-controller/pkg/ovn/ovn.go

```diff
@@ -225,84 +258,187 @@ func getPodNamespacedName(pod *kapi.Pod) string {
 			egressIPTotalTimeout:              config.OVNKubernetesFeature.EgressIPReachabiltyTotalTimeout,
 			egressIPNodeHealthCheckPort:       config.OVNKubernetesFeature.EgressIPNodeHealthCheckPort,
-		},
-		loadbalancerClusterCache:  make(map[kapi.Protocol]string),
-		multicastSupport:          config.EnableMulticast,
-		loadBalancerGroupUUID:     "",
-		aclLoggingEnabled:         true,
-		joinSwIPManager:           nil,
-		retryPods:                 NewRetryObjs(factory.PodType, "", nil, nil, nil),
-		retryNetworkPolicies:      NewRetryObjs(factory.PolicyType, "", nil, nil, nil),
-		retryNodes:                NewRetryObjs(factory.NodeType, "", nil, nil, nil),
-		retryEgressFirewalls:      NewRetryObjs(factory.EgressFirewallType, "", nil, nil, nil),
-		retryEgressIPs:            NewRetryObjs(factory.EgressIPType, "", nil, nil, nil),
-		retryEgressIPNamespaces:   NewRetryObjs(factory.EgressIPNamespaceType, "", nil, nil, nil),
-		retryEgressIPPods:         NewRetryObjs(factory.EgressIPPodType, "", nil, nil, nil),
-		retryEgressNodes:          NewRetryObjs(factory.EgressNodeType, "", nil, nil, nil),
-		retryCloudPrivateIPConfig: NewRetryObjs(factory.CloudPrivateIPConfigType, "", nil, nil, nil),
-		retryNamespaces:           NewRetryObjs(factory.NamespaceType, "", nil, nil, nil),
-		recorder:                  recorder,
-		nbClient:                  libovsdbOvnNBClient,
-		sbClient:                  libovsdbOvnSBClient,
-		svcController:             svcController,
-		svcFactory:                svcFactory,
-		egressSvcController:       egressSvcController,
-		podRecorder:               metrics.NewPodRecorder(),
+		}
+		oc.externalGWCache = make(map[ktypes.NamespacedName]*externalRouteInfo)
+		oc.exGWCacheMutex = sync.RWMutex{}
+
+		oc.retryEgressFirewalls = NewRetryObjs(factory.EgressFirewallType, "", nil, nil, nil)
+		oc.retryEgressIPs = NewRetryObjs(factory.EgressIPType, "", nil, nil, nil)
+		oc.retryEgressIPNamespaces = NewRetryObjs(factory.EgressIPNamespaceType, "", nil, nil, nil)
+		oc.retryEgressIPPods = NewRetryObjs(factory.EgressIPPodType, "", nil, nil, nil)
+		oc.retryEgressNodes = NewRetryObjs(factory.EgressNodeType, "", nil, nil, nil)
+		oc.retryCloudPrivateIPConfig = NewRetryObjs(factory.CloudPrivateIPConfigType, "", nil, nil, nil)
+
+		if addressSetFactory == nil {
+			addressSetFactory = addressset.NewOvnAddressSetFactory(cm.nbClient)
+		}
+		oc.addressSetFactory = addressSetFactory
+		oc.sharedNetpolPortGroups = syncmap.NewSyncMap[*defaultDenyPortGroups]()
+		oc.namespaces = make(map[string]*namespaceInfo)
+		oc.namespacesMutex = sync.Mutex{}
+		oc.retryNamespaces = NewRetryObjs(factory.NamespaceType, "", nil, nil, nil)
+		oc.retryNetworkPolicies = NewRetryObjs(factory.PolicyType, "", nil, nil, nil)
+
+		oc.wg = cm.defaultWg
+		oc.multicastSupport = config.EnableMulticast
+		cm.defaultController = oc
+	} else {
+		oc.wg = &sync.WaitGroup{}
+		oc.multicastSupport = false // disable multicast support for now
```

Comment:
I don't necessarily have a problem with this but I don't think that an OvnController with almost all the stuff uninitialized (like `retryNetworkPolicies`, since it will actually be watching for NPs) would ever work. Would having a "Not implemented" panic be more clear in this respect?
I don't  have a clear picture of how we are going to split the OvnController functionality between what we need for a default network vs a  disconnected L3 topology and the L2 topology.

Reply:
retryNetworkPolicies will not be used for secondary network for now. It will be supported as a separate PR. We will make use of https://github.com/k8snetworkplumbingwg/multi-networkpolicy/blob/master/scheme.yml.